tool-plugins/intellij/src/main/java/io/ballerina/plugins/idea/preloading/LSPUtils.java

```diff
@@ -0,0 +1,143 @@
+import io.ballerina.plugins.idea.sdk.BallerinaSdkUtils;
+import io.ballerina.plugins.idea.settings.autodetect.BallerinaAutoDetectionSettings;
+import io.ballerina.plugins.idea.settings.langserverlogs.LangServerLogsSettings;
+import org.jetbrains.annotations.NotNull;
+import org.wso2.lsp4intellij.IntellijLanguageClient;
+import org.wso2.lsp4intellij.client.languageserver.serverdefinition.ProcessBuilderServerDefinition;
+
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import static io.ballerina.plugins.idea.BallerinaConstants.BAL_FILE_EXT;
+import static io.ballerina.plugins.idea.BallerinaConstants.LAUNCHER_SCRIPT_PATH;
+import static io.ballerina.plugins.idea.BallerinaConstants.SYS_PROP_EXPERIMENTAL;
+import static io.ballerina.plugins.idea.BallerinaConstants.SYS_PROP_LS_DEBUG;
+import static io.ballerina.plugins.idea.BallerinaConstants.SYS_PROP_LS_TRACE;
+
+/**
+ * Language server protocol related utils.
+ */
+public class LSPUtils {
+
+    private static BallerinaAutoDetectNotifier autoDetectNotifier = new BallerinaAutoDetectNotifier();
+
+    /**
+     * Registered language server definition using currently opened ballerina projects.
+     */
+    static void registerServerDefinition() {
+        Project[] openProjects = ProjectManager.getInstance().getOpenProjects();
+        for (Project project : openProjects) {
+            registerServerDefinition(project);
+        }
+    }
+
+    static boolean registerServerDefinition(Project project) {
+
+        boolean autoDetected = false;
+
+        //If the project does not have a ballerina SDK attached, ballerinaSdkPath will be null.
+        BallerinaSdk balSdk = BallerinaSdkUtils.getBallerinaSdkFor(project);
+        String balSdkPath = balSdk.getSdkPath();
+
+        // Checks for the user-configured auto detection settings.
+        if (balSdkPath == null && BallerinaAutoDetectionSettings.getInstance(project).isAutoDetectionEnabled()) {
+
+            //If a ballerina SDK is not configured for the project, Plugin tries to auto detect the ballerina SDK.
+            showInIdeaEventLog(project, String.format("No ballerina SDK is found for project: %s\n " +
+                    "Trying to Auto detect Ballerina Home...", project.getBasePath()));
+
+            balSdkPath = BallerinaSdkUtils.autoDetectSdk(project);
+            autoDetected = true;
+        }
+
+        if (!Strings.isNullOrEmpty(balSdkPath)) {
+            boolean success = doRegister(project, balSdkPath);
+            if (success && autoDetected) {
+                BallerinaPreloadingActivity.LOG.info(String.format("Auto-detected Ballerina Home: %s for the " +
+                        "project: %s", balSdkPath, project.getBasePath()));
+                showInIdeaEventLog(project, "Auto-Detected Ballerina Home: " + balSdkPath);
+            }
+            return success;
+        } else {
+            if (BallerinaAutoDetectionSettings.getInstance(project).isAutoDetectionEnabled()) {
```

Comment:
Is it possible to convert this to an if else clause?

Reply:
Updated